mlflow/utils/proto_json_utils.py

```diff
@@ -1,17 +1,79 @@
 import base64
 import datetime
 
+import json
 from json import JSONEncoder
 
 from google.protobuf.json_format import MessageToJson, ParseDict
+from google.protobuf.descriptor import FieldDescriptor
 
 from mlflow.exceptions import MlflowException
 from collections import defaultdict
+from functools import partial
+
+
+def _mark_int64_fields(proto_message):
+    """Converts a proto message to JSON, preserving only the int64/uint64/fixed64 fields."""
+
+    json_dict = {}
+    for field, value in proto_message.ListFields():
+        if field.type == FieldDescriptor.TYPE_MESSAGE:
+            ftype = partial(_mark_int64_fields)
+        elif field.type in [
+            FieldDescriptor.TYPE_INT64,
+            FieldDescriptor.TYPE_UINT64,
+            FieldDescriptor.TYPE_FIXED64,
+        ]:
+            ftype = int
+        else:
+            # Skip all non-int64 fields.
+            continue
+
+        if field.label == FieldDescriptor.LABEL_REPEATED:
+            json_value = []
+            for v in value:
+                json_value.append(ftype(v))
+        else:
+            json_value = ftype(value)
+
+        json_dict[field.name] = json_value
+    return json_dict
+
+
+def _merge_json_dicts(from_dict, to_dict):
+    """Merges the json elements of from_dict into to_dict. Only works for json dicts converted from proto messages"""
+
+    for key in from_dict:
+        value = to_dict[key]
+        if isinstance(value, dict):
+            _merge_json_dicts(from_dict[key], to_dict[key])
+        elif isinstance(value, list):
+            for i in range(len(value)):
+                if isinstance(value[i], dict):
+                    _merge_json_dicts(from_dict[key][i], to_dict[key][i])
+                else:
+                    to_dict[key][i] = from_dict[key][i]
+        else:
+            to_dict[key] = from_dict[key]
+    return to_dict
 
 
 def message_to_json(message):
     """Converts a message to JSON, using snake_case for field names."""
-    return MessageToJson(message, preserving_proto_field_name=True)
+
+    # Google's MessageToJson API converts int64/fixed64/unit64 proto fields to JSON strings.
```

Comment:
Add context to why, e.g. citing the comment https://github.com/protocolbuffers/protobuf/issues/2954#issuecomment-292671661

Reply:
Good point! Done